js/tablenator.js

```diff
@@ -29,44 +43,90 @@ bcpl.tablenator = (($, _) => {
 		return newTableBodyCollection;
 	};
 
-	const reformat = (tableIndex, tableElement) => {
+	const createMobileTables = (tableIndex, tableElement, callback) => {
 		const $table = $(tableElement);
 		const $headings = $table.find('th');
 		const $dataRows = $table.find('tr').not($headings.closest('tr'));
 
 		const newTableBodyCollection = buildTwoColumnTables($headings, $dataRows);
 
-		newTableBodyCollection.map(($newTable) =>
-			$originalTableCache
-				.eq(tableIndex)
-				.parent()
-				.append($newTable));
+		newTableBodyCollection.reverse()
+			.map(($newTable) => {
+				if (tablenatorOptions.isParentForm) {
+					return $originalTableCache
+						.eq(tableIndex)
+						.closest('form')
+						.after($newTable);
+				}
+				return $originalTableCache
+					.eq(tableIndex)
+					.parent()
+					.append($newTable);
+			});
+
+		$table.hide();
 
-		$table.detach();
+		if (callback && typeof callback === 'function') {
+			callback();
+		}
 	};
 
 	const windowResizehandler = (resizeEvent) => {
 		const windowWidth = $(resizeEvent.target).width();
 
 		if (windowWidth < breakpoint) {
-			$originalTableCache.each(reformat);
+			$originalTableCache.hide();
+			$(`.${responsiveTableClass}`).show();
+
+			callFnOption(tablenatorOptions, 'onMobileSize');
 		} else {
-			const parentCollection = $('.tablenator-responsive-table').map((index, tableElement) => $(tableElement).parent().get());
+			$originalTableCache.show();
+			$(`.${responsiveTableClass}`).hide();
 
-			$.unique(parentCollection).each((index, parentElement) => $(parentElement).empty().append($originalTableCache.eq(index)));
+			callFnOption(tablenatorOptions, 'onDefaultSize');
 		}
 	};
 
-	const init = (tableSelector, screenBreakpoint) => {
+	const hasProperty = (obj, propertyName) => obj && Object.prototype.hasOwnProperty.call(obj, propertyName);
```

Comment:
```
Object.prototype.hasOwnProperty.call(obj, propertyName)
```
Why this and not
```
obj.hasOwnProperty(propertyName)
```

Reply:
proxying @mxsnyder https://stackoverflow.com/a/12017703/1143670